src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.RegEnumKeyEx.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal partial class Interop
+{
+    internal partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, CharSet = CharSet.Unicode, BestFitMapping = false, EntryPoint = "RegEnumKeyExW")]
+        internal static extern unsafe int RegEnumKeyEx(
+            SafeRegistryHandle hKey,
+            int dwIndex,
+            char[] lpName,
+            ref int lpcbName,
+            int[] lpReserved,
+            [Out]StringBuilder lpClass,
```

Comment:
nit: should we use ```out``` keyword here ?

Reply:
`[Out]` is correct here. `out` controls how the argument passed at language/runtime level. `[Out]` controls how the argument is marshaled.